rust-arroyo/src/backends/kafka/types.rs

```diff
@@ -143,4 +151,16 @@ mod tests {
         assert_eq!(base_record.key, Some(&b"key".to_vec()));
         assert_eq!(base_record.payload, Some(&b"message".to_vec()));
     }
+
+    #[test]
+    fn test_headers() {
+        let mut headers = self::Headers::new();
+        headers = headers.insert("key1", Some(b"value1".to_vec()));
+        headers = headers.insert("key2", Some(b"value2".to_vec()));
+        headers = headers.insert("key3", Some(b"value3".to_vec()));
+
+        assert_eq!(headers.get("key1"), Some(b"value1".to_vec()).as_deref());
+        assert_eq!(headers.get("key2"), Some(b"value2".to_vec()).as_deref());
```

Comment:
```suggestion
        assert_eq!(headers.get("key1"), Some(b"value1"));
        assert_eq!(headers.get("key2"), Some(b"value2"));
```

Reply:
The suggested change gives the following error
error[E0308]: mismatched types
   --> rust-arroyo/src/backends/kafka/types.rs:162:41
    |
162 |         assert_eq!(headers.get("key1"), Some(b"value1"));
    |                                         ^^^^^^^^^^^^^^^ expected `Option<&[u8]>`, found `Option<&[u8; 6]>`
    |
    = note: expected enum `Option<&[u8]>`
               found enum `Option<&[u8; 6]>`

error[E0308]: mismatched types
   --> rust-arroyo/src/backends/kafka/types.rs:163:41
    |
163 |         assert_eq!(headers.get("key2"), Some(b"value2"));
    |                                         ^^^^^^^^^^^^^^^ expected `Option<&[u8]>`, found `Option<&[u8; 6]>`
    |
    = note: expected enum `Option<&[u8]>`
               found enum `Option<&[u8; 6]>`
               
Didn't know rust would be so restrictive. Made a change to map the value and return its ref.